js/src/whalesong/statusV3.js

```diff
@@ -0,0 +1,81 @@
+import { command, Iterator } from "../manager";
+import { CollectionManager, ModelManager } from "./common";
+import {
+  MessageManager,
+  MessageCollectionManager
+} from "./message";
+import {
+  ContactManager
+} from './contact.js';
+
+export class StatusV3Manager extends ModelManager {
+  static mapModel(item) {
+    return Object.assign(ModelManager.mapModel(item), {
+      id: item.id._serialized,
+      contact: item.contact ? ContactManager.mapModel(item.contact) : null,
+      msgs: null,
+      expireTs: item.expireTs,
+      hasUnread: item.hasUnread,
+      lastReceivedKey: item.lastReceivedKey ? item.lastReceivedKey._serialized : null,
+      totalCount: item.totalCount,
+      unreadCount: item.unreadCount,
+      readKeys: item.readKeys
+    });
+  }
+
+  constructor(model) {
+    super(model);
+
+    this.addSubmanager('msgs', new MessageCollectionManager(model.msgs));
+    this.addSubmanager('contact', new ContactManager(this.model.contact));
+  }
+
+  @command
+  async sendReadStatus({
+    messageId,
+  }) {
+    return await this.model.sendReadStatus(
+      this.getSubmanager('msgs').loadItem(messageId).id,
+      this.model.id
+    );
+  }
+
+  @command
+  async loadMore() {
+    return this.model.loadMore();
```

Comment:
Is there any way to know if all statuses are already loaded?

Reply:
I'm not sure if there is a way to know if the statuses are loaded, because sometimes, when calling `get_items` right after connected returns an empty array, as a workaround, calling `sync` before calling `get_items` works.

Reply:
This method, I'm really thinking about removing it. In the WhastApp Web, this method is called when you are seeing a Status, then it transition to another one, right in the transition. Maybe it's just the transition animation, or something like this in the Web, but to use in the terminal/console, it's not needed to call this method at all.